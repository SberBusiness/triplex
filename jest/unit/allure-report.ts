import {JasmineAllureReporter} from 'allure-jasmine';
import {Status, TestResult} from 'allure-js-commons';

const reporter = new JasmineAllureReporter({
    resultsDir: 'tests/out/unit/allure/allure-results',
    testMapper: (result: TestResult) => {
        if (result.status == Status.SKIPPED) result.fullName = `(WAS SKIPPED) ${result.fullName}`;
        return result;
    },
});

// @ts-ignore
jasmine.getEnv().addReporter(reporter);

export const allure = reporter.getInterface();
