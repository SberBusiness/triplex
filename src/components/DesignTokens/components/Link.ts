import {TDesignTokenValue, TDesignTokenValues} from '@sberbusiness/triplex/components/DesignTokens/types/DesignTokenTypes';

// Название токенов компонента Link.
export const designTokensComponentsLinkKeys = [
    'BorderColor_Default',
    'BorderColor_Focus',
    'Text_Color_Default',
    'Text_Color_Hover',
    'Text_Color_Active',
    'Line_Color_Default',
    'Line_Color_Hover',
    'Line_Color_Active',
] as const;
// Тип, содержащий названия токенов компонента Link.
export type TDesignTokensComponentsLinkKeys = (typeof designTokensComponentsLinkKeys)[number];
// Тип, содержащий названия токенов компонента Link и их значения.
export type TDesignTokensComponentsLinkValue = Record<TDesignTokensComponentsLinkKeys, TDesignTokenValue>;
// Тип, содержащий названия токенов компонента Link и их значения в светлой и темной теме.
export type TDesignTokensComponentsLinkValues = Record<TDesignTokensComponentsLinkKeys, TDesignTokenValues>;
// Тип локальных токенов компонента Link.
export type TDesignTokensComponentsLink = {Link: TDesignTokensComponentsLinkValue};

// Токены компонента Link в светлой и темной темах.
export const Link_Tokens: TDesignTokensComponentsLinkValues = {
    BorderColor_Default: [{ref: 'Basic.0'}, {ref: 'Basic.0'}], // var(--triplex-Link-BorderColor_Default)
    BorderColor_Focus: [{ref: 'Accent.500'}, {ref: 'Accent.500'}], // var(--triplex-Link-BorderColor_Focus)
    Text_Color_Default: [{ref: 'Info.700'}, {ref: 'Info.500'}], // var(--triplex-Link-Text_Color_Default)
    Text_Color_Hover: [{ref: 'Primary.700'}, {ref: 'Primary.500'}], // var(--triplex-Link-Text_Color_Hover)
    Text_Color_Active: [{ref: 'Info.900'}, {ref: 'Info.700'}], // var(--triplex-Link-Text-Color_Active)
    Line_Color_Default: [{ref: 'Basic.100'}, {ref: 'Basic.700'}], // var(--triplex-Link-Line_Color_Default)
    Line_Color_Hover: [{ref: 'Neutral.900'}, {ref: 'Neutral.900'}], // var(--triplex-Link-Line_Color_Hover)
    Line_Color_Active: [{ref: 'Basic.100'}, {ref: 'Basic.700'}], // var(--triplex-Link-Line_Color_Active)
};
