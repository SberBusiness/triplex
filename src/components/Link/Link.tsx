import React, {useState} from 'react';
import {classnames} from '@sberbusiness/triplex/utils/classnames/classnames';

// Размеры компонента Link.
export enum ELinkSize {
    SM = 'SM',
    LG = 'LG',
}

// Типы компонента Link.
export enum ELinkType {
    TEXT = 'text',
    LINE = 'line',
}

/** Общие свойства компонента Link. */
interface ILinkCommonProps extends React.AnchorHTMLAttributes<HTMLAnchorElement> {
    /** Размер текста. */
    size: ELinkSize;
    /** Тело гиперссылки. */
    children: React.ReactNode;
}

/** Свойства компонента Link типа Text. */
export interface ILinkTextProps extends ILinkCommonProps {
    /** Тип ссылки. */
    linkType: ELinkType.TEXT;
    /** Рендер функция предшествующего контента. */
    contentBefore?: () => JSX.Element;
    /** Рендер функция последующего контента. */
    contentAfter?: () => JSX.Element;
}

/** Свойства компонента Link типа Line. */
export interface ILinkLineProps extends ILinkCommonProps {
    /** Тип ссылки. */
    linkType: ELinkType.LINE;
    /** Рендер функция предшествующего контента. */
    contentBefore?: never;
    /** Рендер функция последующего контента. */
    contentAfter?: never;
}

/** Свойства компонента Link. */
type TLinkProps = ILinkTextProps | ILinkLineProps;

/** Гиперссылка. */
export const Link = React.forwardRef<HTMLAnchorElement, TLinkProps>(
    ({children, className, linkType, size, onBlur, onMouseDown, contentBefore, contentAfter, ...rest}, ref) => {
        const [focusedByClick, setFocusedByClick] = useState(false);

        const handleMouseDown = (event: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => {
            onMouseDown?.(event);
            setFocusedByClick(true);
        };

        const handleBlur = (event: React.FocusEvent<HTMLAnchorElement>) => {
            onBlur?.(event);
            if (event.target !== document.activeElement) {
                setFocusedByClick(false);
            }
        };

        const renderContentBefore = () =>
            contentBefore ? (
                <>
                    {/* Zero-width space необходим для правильного выравнивания контента. */}
                    {'\u200B'}
                    {contentBefore()}
                </>
            ) : null;

        const renderContentAfter = () => (contentAfter ? contentAfter() : null);

        /** Рендерит как простой текст. */
        const renderAsSimpleText = (text: string) => {
            const words = text.split(' ');

            if (words.length < 2 || (words.length < 3 && contentBefore && contentAfter)) {
                const className = classnames('cssClass[wordWithContent]', {
                    'cssClass[before]': Boolean(contentBefore),
                    'cssClass[after]': Boolean(contentAfter),
                });
                return (
                    <span className={className}>
                        {renderContentBefore()}
                        {text}
                        {renderContentAfter()}
                    </span>
                );
            }

            const firstWord = words[0];
            const lastWord = words[words.length - 1];
            const restWords = words.slice(1, -1).join(' ');

            const classNameBefore = classnames('cssClass[wordWithContent]', {
                'cssClass[before]': Boolean(contentBefore),
            });
            const firstNode = contentBefore ? (
                <span className={classNameBefore}>
                    {renderContentBefore()}
                    {firstWord}
                </span>
            ) : (
                firstWord
            );

            const classNameAfter = classnames('cssClass[wordWithContent]', {
                'cssClass[after]': Boolean(contentAfter),
            });
            const lastNode = contentAfter ? (
                <span className={classNameAfter}>
                    {lastWord}
                    {renderContentAfter()}
                </span>
            ) : (
                lastWord
            );

            return (
                <>
                    {firstNode} {restWords} {lastNode}
                </>
            );
        };

        /** Рендерит как React Nodes. */
        const renderAsReactNode = (node: React.ReactNode) => {
            const firstNode = contentBefore ? contentBefore() : null;
            const lastNode = contentAfter ? contentAfter() : null;
            return (
                <>
                    {firstNode}
                    {node}
                    {lastNode}
                </>
            );
        };

        const renderContent = (children: React.ReactNode) =>
            typeof children === 'string' ? renderAsSimpleText(children) : renderAsReactNode(children);

        const content = (contentBefore || contentAfter) && linkType === ELinkType.TEXT ? renderContent(children) : children;

        return (
            <a
                ref={ref}
                tabIndex={0}
                role="link"
                {...rest}
                className={classnames(className, 'cssClass[link]', 'hoverable', {
                    'cssClass[sm]': size === ELinkSize.SM,
                    'cssClass[lg]': size === ELinkSize.LG,
                    'cssClass[text]': linkType === ELinkType.TEXT,
                    'cssClass[line]': linkType === ELinkType.LINE,
                    'cssClass[focusVisible]': !focusedByClick,
                })}
                onBlur={handleBlur}
                onMouseDown={handleMouseDown}
            >
                {content}
            </a>
        );
    }
);

Link.displayName = 'Link';
