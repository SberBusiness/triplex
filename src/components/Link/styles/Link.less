@import 'src/styles/style';

.link {
    cursor: pointer;
    padding: 0 2px;
    margin: -1px -3px;
    border: 1px solid var(--triplex-Link-BorderColor_Default);

    &:focus {
        outline: none;

        &.focusVisible {
            border-color: var(--triplex-Link-BorderColor_Focus);
        }
    }
}

.sm {
    .font-size(@font-size-12);
}

.lg {
    .font-size(@font-size-14);
}

.text {
    position: relative;
    color: var(--triplex-Link-Text_Color_Default);
    text-decoration: none;

    &:hover {
        color: var(--triplex-Link-Text_Color_Hover);
    }

    &:active {
        color: var(--triplex-Link-Text_Color_Active);
    }
}

.line {
    color: var(--triplex-Link-Line_Color_Default);
    text-decoration: underline;

    &:hover {
        color: var(--triplex-Link-Line_Color_Hover)
    }

    &:active {
        color: var(--triplex-Link-Line_Color_Active);
    }
}

.wordWithContent {
    white-space: nowrap;
    display: inline-flex;
    align-items: center;
}

.before :first-child {
    margin-right: @metric-4;
}

.after :last-child {
    margin-left: @metric-4;
}
