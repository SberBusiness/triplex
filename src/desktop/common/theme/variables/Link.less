@import ~'@sberbusiness/core-styles/colors.less';

:root {
    --link-color: @color-sky-02;
    --link-color-hover: @color-malachite-02;
    --link-color-active: @color-sky-01;
    --line-color: @color-snowy;
    --line-color-hover: @color-asphalt-01;
    --line-color-active: @color-snowy;
}

.triplex-theme-dark {
    --link-color: @color-sky-03;
    --link-color-hover: @color-malachite-03;
    --link-color-active: @color-sky-02;
    --line-color: @color-graphite-01;
    --line-color-hover: @color-asphalt-01;
    --line-color-active: @color-graphite-01;
}
