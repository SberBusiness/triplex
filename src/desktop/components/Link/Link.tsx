import {classnames} from '@sberbusiness/triplex/common/utils/classnames/classnames';
import React, {useState} from 'react';

export enum ELinkSize {
    SM = 'SM',
    LG = 'LG',
}

export enum ELinkType {
    TEXT = 'text',
    LINE = 'line',
}

/** Свойства компонента Link. */
interface ILinkCommonProps extends React.AnchorHTMLAttributes<HTMLAnchorElement> {
    /** Размер текста. */
    size: ELinkSize;
    /** Тело гиперссылки. */
    children: React.ReactNode;
}

export interface ILinkTextProps extends ILinkCommonProps {
    /** Гиперссылка без подчеркивания. */
    linkType: ELinkType.TEXT;
    /** Рендер функция предшествующего контента. */
    contentBefore?: () => JSX.Element;
    /** Рендер функция последующего контента. */
    contentAfter?: () => JSX.Element;
}

export interface ILinkLineProps extends ILinkCommonProps {
    /** Гиперссылка с подчеркиванием. */
    linkType: ELinkType.LINE;
    /** Рендер функция предшествующего контента. */
    contentBefore?: never;
    /** Рендер функция последующего контента. */
    contentAfter?: never;
}

type TLinkProps = ILinkTextProps | ILinkLineProps;

/** Компонент гиперссылки. */
export const Link = React.forwardRef<HTMLAnchorElement, TLinkProps>(
    ({children, className, linkType, size, onBlur, onMouseDown, contentBefore, contentAfter, ...rest}, ref) => {
        const [focusedByClick, setFocusedByClick] = useState(false);

        const handleMouseDown = (event: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => {
            onMouseDown?.(event);
            setFocusedByClick(true);
        };

        const handleBlur = (event: React.FocusEvent<HTMLAnchorElement>) => {
            onBlur?.(event);
            if (event.target !== document.activeElement) {
                setFocusedByClick(false);
            }
        };

        const renderContentBefore = () =>
            contentBefore ? (
                <>
                    {/* Zero-width space необходим для правильного выравнивания контента. */}
                    {'\u200B'}
                    {contentBefore()}
                </>
            ) : null;

        const renderContentAfter = () => (contentAfter ? contentAfter() : null);

        /** Рендерит как простой текст. */
        const renderAsSimpleText = (text: string) => {
            const words = text.split(' ');

            if (words.length < 2 || (words.length < 3 && contentBefore && contentAfter)) {
                const className = classnames('cssClass[wordWithContent]', {
                    'cssClass[before]': Boolean(contentBefore),
                    'cssClass[after]': Boolean(contentAfter),
                });
                return (
                    <span className={className}>
                        {renderContentBefore()}
                        {text}
                        {renderContentAfter()}
                    </span>
                );
            }

            const firstWord = words[0];
            const lastWord = words[words.length - 1];
            const restWords = words.slice(1, -1).join(' ');

            const classNameBefore = classnames('cssClass[wordWithContent]', {
                'cssClass[before]': Boolean(contentBefore),
            });
            const firstNode = contentBefore ? (
                <span className={classNameBefore}>
                    {renderContentBefore()}
                    {firstWord}
                </span>
            ) : (
                firstWord
            );

            const classNameAfter = classnames('cssClass[wordWithContent]', {
                'cssClass[after]': Boolean(contentAfter),
            });
            const lastNode = contentAfter ? (
                <span className={classNameAfter}>
                    {lastWord}
                    {renderContentAfter()}
                </span>
            ) : (
                lastWord
            );

            return (
                <>
                    {firstNode} {restWords} {lastNode}
                </>
            );
        };

        /** Рендерит как React Nodes. */
        const renderAsReactNode = (node: React.ReactNode) => {
            const firstNode = contentBefore ? contentBefore() : null;
            const lastNode = contentAfter ? contentAfter() : null;
            return (
                <>
                    {firstNode}
                    {node}
                    {lastNode}
                </>
            );
        };

        const renderContent = (children: React.ReactNode) =>
            typeof children === 'string' ? renderAsSimpleText(children) : renderAsReactNode(children);

        const content = (contentBefore || contentAfter) && linkType === ELinkType.TEXT ? renderContent(children) : children;

        return (
            <a
                ref={ref}
                tabIndex={0}
                role="link"
                {...rest}
                className={classnames(className, 'cssClass[link]', 'hoverable', {
                    'cssClass[sm]': size === ELinkSize.SM,
                    'cssClass[lg]': size === ELinkSize.LG,
                    'cssClass[text]': linkType === ELinkType.TEXT,
                    'cssClass[line]': linkType === ELinkType.LINE,
                    'cssClass[focusVisible]': !focusedByClick,
                })}
                onBlur={handleBlur}
                onMouseDown={handleMouseDown}
            >
                {content}
            </a>
        );
    }
);

Link.displayName = 'Link';
