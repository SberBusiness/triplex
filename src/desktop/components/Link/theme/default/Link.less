@import '../../../../common/theme/default';
@import '../../../../common/theme/variables/Link';

.link {
    cursor: pointer;
    padding: 0 2px;
    margin: -1px -3px;
    border: 1px solid transparent;

    &:focus {
        outline: none;

        &.focusVisible {
            border: 1px solid @color-sunny-01;
        }
    }
}

.sm {
    .font-size(@font-size-12);
}

.lg {
    .font-size(@font-size-14);
}

.text {
    position: relative;
    color: @color-sky-02;
    color: var(--link-color);
    text-decoration: none;

    &:hover {
        color: @color-malachite-02;
        color: var(--link-color-hover);
    }

    &:active {
        color: @color-sky-01;
        color: var(--link-color-active);
    }
}

.line {
    color: @color-snowy;
    color: var(--line-color);
    text-decoration: underline;

    &:hover:not(:active) {
        color: @color-asphalt-01;
        color: var(--line-color-hover);
    }
}

.wordWithContent {
    white-space: nowrap;
    display: inline-flex;
    align-items: center;
}

.before :first-child {
    margin-right: @metric-4;
}

.after :last-child {
    margin-left: @metric-4;
}
