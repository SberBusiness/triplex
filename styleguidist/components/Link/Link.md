```jsx {"props": {"className": "white-to-black-background"}}
import {ELinkSize, ELinkType} from '@sberbusiness/triplex/components/Link/Link';

const handleClick = (event) => event.preventDefault();

<div style={{display: 'flex'}}>
    <div style={{flexBasis: '50%', textAlign: 'center'}}>
        <Link href="#" linkType={ELinkType.TEXT} size={ELinkSize.LG} onClick={handleClick}>
            Link Text
        </Link>
    </div>
    <div style={{flexBasis: '50%', textAlign: 'center'}}>
        <Link href="#" linkType={ELinkType.LINE} size={ELinkSize.LG} onClick={handleClick}>
            Link Line
        </Link>
    </div>
</div>
```

### Small size

```jsx {"props": {"className": "white-to-black-background"}}
import {ELinkSize, ELinkType} from '@sberbusiness/triplex/components/Link/Link';

const handleClick = (event) => event.preventDefault();

<div style={{display: 'flex'}}>
    <div style={{flexBasis: '50%', textAlign: 'center'}}>
        <Link href="#" linkType={ELinkType.TEXT} size={ELinkSize.SM} onClick={handleClick}>
            Link Text
        </Link>
    </div>
    <div style={{flexBasis: '50%', textAlign: 'center'}}>
        <Link href="#" linkType={ELinkType.LINE} size={ELinkSize.SM} onClick={handleClick}>
            Link Line
        </Link>
    </div>
</div>
```

### External

```jsx
import {ELinkSize, ELinkType} from '@sberbusiness/triplex/components/Link/Link';
import {LinkNavIcon16} from '@sberbusiness/icons/LinkNavIcon16';

const handleClick = (event) => event.preventDefault();

const renderContentAfter = () => <LinkNavIcon16 />;

<Link
    href="#"
    linkType={ELinkType.TEXT}
    size={ELinkSize.LG}
    contentAfter={renderContentAfter}
    onClick={handleClick}
>
    Link Text
</Link>
```
