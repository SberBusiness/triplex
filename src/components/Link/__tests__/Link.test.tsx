import {shallow} from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import {ELinkSize, ELinkType, Link} from '@sberbusiness/triplex/components/Link/Link';
import {allure} from '@jest/unit/allure-report';

describe('Link', () => {
    beforeEach(() => {
        allure.feature('Link');
    });

    it('renders correctly', () => {
        const tree = renderer
            .create(
                <Link size={ELinkSize.SM} linkType={ELinkType.TEXT} href="#">
                    Test
                </Link>
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders correctly with after content', () => {
        const tree = renderer
            .create(
                <Link size={ELinkSize.SM} linkType={ELinkType.TEXT} contentAfter={() => <span>mock</span>} href="#">
                    Test
                </Link>
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders correctly with two words and after content', () => {
        const tree = renderer
            .create(
                <Link size={ELinkSize.SM} linkType={ELinkType.TEXT} contentAfter={() => <span>mock</span>} href="#">
                    Test link
                </Link>
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders correctly with before content', () => {
        const tree = renderer
            .create(
                <Link size={ELinkSize.SM} linkType={ELinkType.TEXT} contentBefore={() => <span>mock</span>} href="#">
                    Test
                </Link>
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders correctly with two words and before content', () => {
        const tree = renderer
            .create(
                <Link size={ELinkSize.SM} linkType={ELinkType.TEXT} contentBefore={() => <span>mock</span>} href="#">
                    Test link
                </Link>
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders correctly with both before and after content', () => {
        const tree = renderer
            .create(
                <Link
                    size={ELinkSize.SM}
                    linkType={ELinkType.TEXT}
                    contentBefore={() => <span>mock</span>}
                    contentAfter={() => <span>mock</span>}
                    href="#"
                >
                    Test
                </Link>
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders correctly with two words and both before and after content', () => {
        const tree = renderer
            .create(
                <Link
                    size={ELinkSize.SM}
                    linkType={ELinkType.TEXT}
                    contentBefore={() => <span>mock</span>}
                    contentAfter={() => <span>mock</span>}
                    href="#"
                >
                    Test link
                </Link>
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders correctly with three words and both before and after content', () => {
        const tree = renderer
            .create(
                <Link
                    size={ELinkSize.SM}
                    linkType={ELinkType.TEXT}
                    contentBefore={() => <span>mock</span>}
                    contentAfter={() => <span>mock</span>}
                    href="#"
                >
                    Test link again
                </Link>
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('triggers click event', () => {
        const handleClick = jest.fn();
        const link = shallow(
            <Link size={ELinkSize.LG} linkType={ELinkType.TEXT} href="#" onClick={handleClick}>
                Test link
            </Link>
        );
        link.simulate('click');

        expect(handleClick).toBeCalledTimes(1);
    });

    it('triggers mouseOver event', () => {
        const handleMouseOver = jest.fn();
        const link = shallow(
            <Link size={ELinkSize.LG} linkType={ELinkType.TEXT} href="#" onMouseOver={handleMouseOver}>
                Test link
            </Link>
        );
        link.simulate('mouseOver');

        expect(handleMouseOver).toBeCalledTimes(1);
    });

    it('triggers mouseOut event', () => {
        const handleMouseOut = jest.fn();
        const link = shallow(
            <Link size={ELinkSize.LG} linkType={ELinkType.TEXT} href="#" onMouseOut={handleMouseOut}>
                Test link
            </Link>
        );
        link.simulate('mouseOut');

        expect(handleMouseOut).toBeCalledTimes(1);
    });

    it('triggers mouseDown event', () => {
        const handleMouseDown = jest.fn();
        const link = shallow(
            <Link size={ELinkSize.LG} linkType={ELinkType.TEXT} href="#" onMouseDown={handleMouseDown}>
                Test link
            </Link>
        );
        link.simulate('mouseDown');

        expect(handleMouseDown).toBeCalledTimes(1);
    });

    it('triggers mouseUp event', () => {
        const handleMouseUp = jest.fn();
        const link = shallow(
            <Link size={ELinkSize.LG} linkType={ELinkType.TEXT} href="#" onMouseUp={handleMouseUp}>
                Test link
            </Link>
        );
        link.simulate('mouseUp');

        expect(handleMouseUp).toBeCalledTimes(1);
    });
});
