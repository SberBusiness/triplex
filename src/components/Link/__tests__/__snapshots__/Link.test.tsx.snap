// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders correctly 1`] = `
<a
  className="cssClass[link] hoverable cssClass[focusVisible] cssClass[sm] cssClass[text]"
  data-tx="10.0.0"
  href="#"
  onBlur={[Function]}
  onMouseDown={[Function]}
  role="link"
>
  Test
</a>
`;

exports[`Link renders correctly with after content 1`] = `
<a
  className="cssClass[link] hoverable cssClass[focusVisible] cssClass[sm] cssClass[text]"
  data-tx="10.0.0"
  href="#"
  onBlur={[Function]}
  onMouseDown={[Function]}
  role="link"
>
  <span
    className="cssClass[wordWithContent] cssClass[after]"
  >
    Test
    <span>
      mock
    </span>
  </span>
</a>
`;

exports[`Link renders correctly with before content 1`] = `
<a
  className="cssClass[link] hoverable cssClass[focusVisible] cssClass[sm] cssClass[text]"
  data-tx="10.0.0"
  href="#"
  onBlur={[Function]}
  onMouseDown={[Function]}
  role="link"
>
  <span
    className="cssClass[wordWithContent] cssClass[before]"
  >
    ​
    <span>
      mock
    </span>
    Test
  </span>
</a>
`;

exports[`Link renders correctly with both before and after content 1`] = `
<a
  className="cssClass[link] hoverable cssClass[focusVisible] cssClass[sm] cssClass[text]"
  data-tx="10.0.0"
  href="#"
  onBlur={[Function]}
  onMouseDown={[Function]}
  role="link"
>
  <span
    className="cssClass[wordWithContent] cssClass[after] cssClass[before]"
  >
    ​
    <span>
      mock
    </span>
    Test
    <span>
      mock
    </span>
  </span>
</a>
`;

exports[`Link renders correctly with three words and both before and after content 1`] = `
<a
  className="cssClass[link] hoverable cssClass[focusVisible] cssClass[sm] cssClass[text]"
  data-tx="10.0.0"
  href="#"
  onBlur={[Function]}
  onMouseDown={[Function]}
  role="link"
>
  <span
    className="cssClass[wordWithContent] cssClass[before]"
  >
    ​
    <span>
      mock
    </span>
    Test
  </span>
   
  link
   
  <span
    className="cssClass[wordWithContent] cssClass[after]"
  >
    again
    <span>
      mock
    </span>
  </span>
</a>
`;

exports[`Link renders correctly with two words and after content 1`] = `
<a
  className="cssClass[link] hoverable cssClass[focusVisible] cssClass[sm] cssClass[text]"
  data-tx="10.0.0"
  href="#"
  onBlur={[Function]}
  onMouseDown={[Function]}
  role="link"
>
  Test
   
  
   
  <span
    className="cssClass[wordWithContent] cssClass[after]"
  >
    link
    <span>
      mock
    </span>
  </span>
</a>
`;

exports[`Link renders correctly with two words and before content 1`] = `
<a
  className="cssClass[link] hoverable cssClass[focusVisible] cssClass[sm] cssClass[text]"
  data-tx="10.0.0"
  href="#"
  onBlur={[Function]}
  onMouseDown={[Function]}
  role="link"
>
  <span
    className="cssClass[wordWithContent] cssClass[before]"
  >
    ​
    <span>
      mock
    </span>
    Test
  </span>
   
  
   
  link
</a>
`;

exports[`Link renders correctly with two words and both before and after content 1`] = `
<a
  className="cssClass[link] hoverable cssClass[focusVisible] cssClass[sm] cssClass[text]"
  data-tx="10.0.0"
  href="#"
  onBlur={[Function]}
  onMouseDown={[Function]}
  role="link"
>
  <span
    className="cssClass[wordWithContent] cssClass[after] cssClass[before]"
  >
    ​
    <span>
      mock
    </span>
    Test link
    <span>
      mock
    </span>
  </span>
</a>
`;
